.github/workflows/teamcity-services-diff-check.yml

```diff
@@ -0,0 +1,93 @@
+name: TeamCity Services Diff Check
+permissions: read-all
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - '.github/workflows/teamcity-services-diff-check.yml'
+      - 'mmv1/third_party/terraform/.teamcity/components/inputs/services_ga.kt'
+      - 'mmv1/third_party/terraform/.teamcity/components/inputs/services_beta.kt'
+      - 'mmv1/products/**'
```

Comment:
I'm glad we established that the use of paths here + the `git diff` step will help us identify a new service being added pretty accurately. Though after thinking that through I realised that another edge case is probably when a service is promoted to GA. In that situation services_ga.kt will be mis-matched with the GA provider but this GHA won't have an opportunity to be triggered automatically.

I think a solution to this is adding a [schedule](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) trigger and making it run once a week. [This is a useful website](https://crontab.guru/) to check the cron expression. I'd recommend running this on early Tuesday mornings (i.e. sometime ~4am UTC) because that's the last time we could catch a newly-added service missing from TeamCity before the the tests run on Wednesday morning and are used for the release cut.

Reply:
Given the PR-specific stuff below using `git diff`, it might be that we can't add a schedule trigger. Instead we might need to pull out some of the stuff below into reusable workflows, and then create a PR-triggered and schedule-triggered version of the workflow that both use the TeamCity-checking steps.

Reply:
See also: https://github.com/hashicorp/terraform-provider-google/issues/15633#issuecomment-1961697734

Reply:
Latest commit adds an almost identical workflow that runs on PRs, only difference is the removal of `check-pr` job and the addition of `schedule` where it'll run every tuesday morning (4am UTC)

Reply:
Looks good! A future improvement could be to use reusable workflows to reduce repetition, but lets leave that for now and get this merged 🚢